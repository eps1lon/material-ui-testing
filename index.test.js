import React from "react";
import TestRenderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import Button from "@material-ui/core/Button";
import Modal from "@material-ui/core/Modal";
import MenuList from "@material-ui/core/MenuList";
import MenuItem from "@material-ui/core/MenuItem";

function createNodeMock() {
  return {
    nodeType: 1, // to trick findDOMNode
    // for focus-visible polyfill
    ownerDocument: {
      addEventListener: () => {},
      removeEventListener: () => {}
    },
    focus: () => {}
  };
}

it("can mount a button", () => {
  const renderer = TestRenderer.create(<Button>test</Button>, {
    createNodeMock
  });
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("can mount a MenuList", () => {
  const renderer = TestRenderer.create(
    <MenuList autoFocus>
      <MenuItem selected>1st item</MenuItem>
    </MenuList>,
    {
      createNodeMock
    }
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});

it("throws with a modal", () => {
  expect(() =>
    TestRenderer.create(
      <Modal open>
        <div />
      </Modal>
    )
  ).toThrow();
});

it("can shallow render a button", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Button />);
  const result = renderer.getRenderOutput();
});
