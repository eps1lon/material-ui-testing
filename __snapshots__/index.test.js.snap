// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can mount a MenuList 1`] = `
<ul
  className="MuiList-root MuiList-padding"
  onKeyDown={[Function]}
  role="menu"
  tabIndex={0}
>
  <li
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button Mui-selected MuiMenuItem-root Mui-selected MuiMenuItem-gutters"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    tabIndex={-1}
  >
    1st item
  </li>
</ul>
`;

exports[`can mount a button 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    test
  </span>
</button>
`;
